app/src/org/commcare/appupdate/CommcareFlexibleAppUpdateManager.java

```diff
@@ -0,0 +1,206 @@
+        try {
+            boolean success = mAppUpdateManager.startUpdateFlowForResult(
+                    mAppUpdateInfo, AppUpdateType.FLEXIBLE, activity, IN_APP_UPDATE_REQUEST_CODE);
+            if (!success) {
+                Logger.log(TAG, "startUpdate|requested update cannot be started");
+            }
+        } catch (IntentSender.SendIntentException exception) {
+            mInstallStatus = InstallStatus.FAILED;
+            Logger.log(TAG, "startUpdate|failed with : " + exception.getMessage());
+        }
+    }
+
+    @NonNull
+    @Override
+    public AppUpdateState getStatus() {
+        return mAppUpdateState;
+    }
+
+    @Override
+    public void completeUpdate() {
+        mAppUpdateManager.completeUpdate()
+                .addOnSuccessListener(result -> {
+                    Logger.log(TAG, "completeUpdate|was successful");
+                    publishStatus();
+                })
+                .addOnFailureListener(exception -> {
+                    Logger.log(TAG, "completeUpdate|failed with : " + exception.getMessage());
+                    publishStatus();
+                });
+    }
+
+    @Nullable
+    @Override
+    public Integer getProgress() {
+        if (mInstallStatus != InstallStatus.DOWNLOADING) {
+            return null;
+        }
+        return percentageDownloaded;
+    }
+
+    @Override
+    public int getErrorCode() {
+        return mInstallErrorCode;
+    }
+
+    @Override
+    public void skipVersion() {
+        if (mAppUpdateInfo == null) {
+            return;
+        }
+        int currentAvailableVersion = mAppUpdateInfo.availableVersionCode();
+        HiddenPreferences.skipCommCareVersionForUpdate(currentAvailableVersion);
+    }
+
+    //endregion
+
+    @Override
+    public void onStateUpdate(InstallState state) {
+        Logger.log(TAG, "Install state updated with install status: " + state.installStatus()
+                + ", and error code: " + state.installErrorCode());
+        mInstallErrorCode = state.installErrorCode();
+        if (state.installStatus() == InstallStatus.DOWNLOADING) {
+            // TODO: Should be keep on publishing status for showing download percentage.
+            // From https://stackoverflow.com/a/10415638/6671572
+            percentageDownloaded = (int) (state.bytesDownloaded() * 100.0 / state.totalBytesToDownload() + 0.5) ;
+        }
+        if (mInstallStatus != state.installStatus()) {
+            mInstallStatus = state.installStatus();
+            publishStatus();
+        }
+    }
+
+    //region Private helpers
+    private void fetchAppUpdateInfo() {
+        mAppUpdateManager.getAppUpdateInfo()
+                .addOnSuccessListener(info -> {
+                    mAppUpdateInfo = info;
+                    mUpdateAvailability = info.updateAvailability();
+                    mInstallStatus = info.installStatus();
+                    Logger.log(TAG, "fetchAppUpdateInfo|success with UpdateAvailability : "
```

Comment:
We can omit the success logs here as this will be triggered every time user comes on Home screen.